### DISCUSSION TITLE
New ESP8266 module, does not seem to want to talk via COM Port/USB or UART via FTDI-USB?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#10339] [General]
New ESP8266 module, does not seem to want to talk via COM Port/USB or UART via FTDI-USB?

I have tried to flash the module via Arduino IDE as well as using Thorny.  Both attempts serial COM port or UART time out results.

esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header.

If I open serial console, I can type AT commands.  And power on the module reports READY.  But no joy beyond that point.  This module is 1 of 3 I purchased, the other two flashed MicroPython without issue.

AT+GMR
AT version:1.1.0.0(May 11 2016 18:09:56)
SDK version:1.5.4(baaeaebb)
compile time:May 20 2016 15:08:19
OK

Since it is responding to AT commands it is not bricked, but not sure what else to try?  Is there a way to force the module via AT commands to accept a flash update?

**Comment by Jibun-no-Kage** [comment]
Bump... No one?

**Reply by robert-hh** [reply]
It looks as if it does not switch to bootloader mode (Apply reset with GPIO0 and Gnd). Does the module look in any way different than the other ones? Part missing?  Bad soldering?

**Comment by Jibun-no-Kage** [comment]
I can't see any difference, but I don't have a microscope or a very strong magnifier to validate.  I doubled checked the pin soldering, that looks to be solid and correct on the suspect module.  I did jumper GND to GPIO0, and tried that right off, did not work, the module just seems to refuse to behave.  I will have time this weekend to test again, before I bin the thing, for salvage parts.  Unfortunately, past modules have been solid from the vendor, so I did not test these immediately, so I may have to eat the $4 since I can't return it this late.  Good news I guess, is the replace order the module was only $3, price came down a bit, guess the ESP32 is dropping the demand for the older ESP8266 variants.

**Reply by Jibun-no-Kage** [reply]
Ok, I happen to find a comment in another forum via Google, where someone has the same issue as I do with this troubled module.  They connected GPIO0 to GND, which I knew about.  But what I did not recall or maybe did not know, was that connecting GPIO2 to 3.3v is sometimes also required.  I did this, and the module responded and allowed flash of MP firmware.  I happened to be using Thonny, so in the REPL I did the firmware check:

>>> import esp
>>> esp.check_fw()
size: 634828
md5: 14a0d953abf53c8d087db862b5ed7d32
True
>>> 

The original link is below...
https://www.esp8266.com/viewtopic.php?f=13&t=21635

Please ignore the 'evil' references to using Arduino IDE... LOL!

**Reply by scruss** [reply]
> ... so I may have to eat the $4 since I can't return it this late

Have you been buying cheap off-brand modules [again](https://forum.micropython.org/viewtopic.php?f=2&t=12948)? Seriously, that game's not worth the candle. Certainly not worth bumping a thread over (please never do that anyway, it's rude).

Take, for example, the [LOLIN D1 Mini V4.0.0](https://www.aliexpress.com/item/32529101036.html). It's US$3.70 direct from the manufacturer (who even provides [support documentation](https://www.wemos.cc/en/latest/d1/d1_mini.html)). It comes flashed with MicroPython. It comes with 4 MB flash and (yay!) a proper USB-C connector. It's nicely made. I've never had a bad one (yet).

But you can get an ESP32-C3 for even less: [WeAct ESP32-C3FH4](https://www.aliexpress.com/item/1005004960064227.html).

**Comment by robert-hh** [comment]
Still interesting why these signals were not applied by esptool.py. There must be something different with the board.

**Comment by Jibun-no-Kage** [comment]
Nope in this case I purchased legit modules.  Just happen to get one it is a bit odd.  It happens.  I wanted the older USB socket to keep an existing standard, but other than that, nothing special.  These are for a solar battery/gimbal/tracking project so the ESP32 would have been major over kill.  Building devices that track the Sun from Sunrise to Sunset.
